source/common/router/upstream_request.cc

```diff
@@ -420,6 +420,11 @@ void UpstreamRequest::onPoolReady(
   stream_info_.setUpstreamSslConnection(info.downstreamSslConnection());
   parent_.callbacks()->streamInfo().setUpstreamSslConnection(info.downstreamSslConnection());
 
+  if (info.upstreamConnectionId().has_value()) {
```

Comment:
Really good point! Thanks for pointing out.

I updated. Let me know if you think this is an appropriate place. Related I had to expose stream info for upstream request. Not sure if it was intentional that we did not have a public API for that.

Reply:
yeah AFIK we were just adding APIs as needed. The only thing I think you should do, as we're in an area where upstream and downstream are intersecting, is comment or rename to make it clear which stream info is being accessed :-)

Reply:
@alyssawilk I was actually confused when I first read the ssl info lines above, and thought there was a bug. In my mind because we are receiving information about a connection that acts as a client connection the term Upstream made sense 😅. I added a comment that is hopefully clarifying. Lmk what you think